crypto/src/ecdsa.rs

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::traits::{
+    Authenticator, EncodeDecodeBase64, KeyPair, SigningKey, ToFromBytes, VerifyingKey,
+};
+use base64ct::{Base64, Encoding};
+use once_cell::sync::OnceCell;
+use serde::de::Error as SerdeError;
+use serde::{de, Deserialize, Serialize};
+use serde_bytes::ByteBuf as SerdeByteBuf;
+use signature::{Signature, Signer, Verifier};
+use std::fmt::{self, Debug, Display};
+
+pub const ECDSA_PRIVATE_KEY_LENGTH: usize = 32;
+pub const ECDSA_PUBLIC_KEY_LENGTH: usize = 33;
+pub const ECDSA_SIGNATURE_LENGTH: usize = 64;
+
+#[readonly::make]
+#[derive(Debug, Clone)]
+pub struct EcdsaPublicKey {
```

Comment:
I'd probably be more specific about the curve that we are using - perhaps Francois has more insight into that.

Reply:
If I'm not mistaken, we might be looking to implement secp256r1 as well.